web-api/src/ingestion/routes.rs

```diff
@@ -115,7 +114,48 @@ struct IngestedDocument {
     snippet: String,
     properties: DocumentProperties,
     tags: Vec<DocumentTag>,
-    is_candidate: bool,
+    is_candidate_op: IsCandidateOp,
+}
+
+#[derive(Clone, Debug, Copy)]
+enum IsCandidateOp {
+    SetTo(bool),
+    DefaultTo(bool),
+}
+
+impl IsCandidateOp {
+    /// Returns the new value and `has_existing && is_changed_value`
+    fn resolve(self, existing: Option<bool>) -> NewIsCandidate {
+        match self {
+            IsCandidateOp::SetTo(new) => NewIsCandidate {
+                value: new,
+                existing_and_has_changed: existing
+                    .map(|previous| previous != new)
+                    .unwrap_or_default(),
+            },
+            IsCandidateOp::DefaultTo(default) => NewIsCandidate {
+                value: existing.unwrap_or(default),
+                existing_and_has_changed: false,
+            },
+        }
+    }
```

Comment:
This is slightly more complex then in the previous version but allows us to only add/remove is candidate where needed for changed documents.